pkg/database/database.go

```diff
@@ -38,6 +41,10 @@ type Database struct {
 
 	// secretManager is used to resolve secrets.
 	secretManager secrets.SecretManager
+
+	// Redis configuration.
+	redisMu   sync.RWMutex
+	redisPool *redis.Pool
```

Comment:
This is a little bit weird now because we already have a cache that's in-memory (see cache.Cache above), and now we have a remote cache via Redis...